src/db/pool.cr

```diff
@@ -47,6 +60,29 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    record Settings,
+      initial_pool_size : Int32,
+      max_pool_size : Int32,
+      max_idle_pool_size : Int32,
+      checkout_timeout : Float64,
+      retry_attempts : Int32,
+      retry_delay : Float64,
+      connection_ttl : Float64,
+      reaper_retry : Float64
+
+    def settings
+      Settings.new(
```

Comment:
This is probably overkill -- we only need two settings to be exposed to the connection. Can just add those two methods, most likely.